src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -251,12 +251,32 @@ export async function run(
   // A cache to hold built shader pipelines.
   const pipelineCache = new Map<String, GPUComputePipeline>();
 
-  // Submit all the cases in batches, each in a separate error scope.
-  const checkResults: Array<Promise<void>> = [];
+  // Submit all the cases in batches, rate-limiting to ensure not too many
+  // batches are in flight simultaneously.
+  const maxBatchesInFlight = 5;
```

Comment:
this number was chosen experimentally based on making it as small as I could without impacting total wall time of the test